waiter/integration/waiter/token_request_test.clj

```diff
@@ -140,46 +143,62 @@
           (let [{:keys [body] :as response} (get-token router-url token :cookies cookies)]
             (assert-response-status response 404)
             (is (str/includes? (str body) "couldn't find token") (str body)))
+          (let [{:keys [body] :as response} (get-token router-url token :cookies cookies :query-params {"show-deleted" true})]
+            (assert-response-status response 200)
+            (let [token-description (parse-token-description body)]
+              (is (contains? token-description :last-update-time))
+              (is (= {:deleted true :health-check-url "/custom-endpoint" :name service-id-prefix :owner (retrieve-username)}
+                     (dissoc token-description :last-update-time)))))
           (let [{:keys [body] :as tokens-response} (list-tokens router-url current-user :cookies cookies)
                 tokens (json/read-str body)]
             (assert-response-status tokens-response 200)
-            (is (not-any? (fn [token-entry] (= token (get token-entry "token"))) tokens))))))))
+            (is (not-any? (fn [token-entry] (= token (get token-entry "token"))) tokens)))))
+      (log/info "deleting the tokens in erase mode")
+      (doseq [token tokens-to-create]
+        (delete-token-and-assert waiter-url token :query-params {"erase" true}))
+      (log/info "ensuring tokens can no longer be retrieved on each router with show-deleted parameter after erase")
+      (doseq [token tokens-to-create]
+        (doseq [[_ router-url] (routers waiter-url)]
+          (let [{:keys [body] :as response} (get-token router-url token :cookies cookies :query-params {"show-deleted" true})]
+            (assert-response-status response 404)
+            (is (str/includes? (str body) "couldn't find token") (str body))))))))
```

Comment:
This test is getting pretty big. Should we break it up into multiple smaller ones?